File: backend/src/main/java/com/example/bookacourt/dtos/BookingStats.java
package com.example.bookacourt.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class BookingStats {
    private int totalBookings;
    private int upcomingBookingsCount;
    private List<BookingRequest> upcomingBookings;
    private int completedBookings;
    private int cancelledBookings;
    private double thisMonthSpent;
}

File: frontend/src/dtos/BookingStats.ts
import type { BookingRequest } from "./Booking";

export interface BookingStats {
  totalBookings: number;
  upcomingBookingsCount: number;
  upcomingBookings: BookingRequest[];
  completedBookings: number;
  cancelledBookings: number;
  thisMonthSpent: number;
}

export interface BookingStatsAdmin {
  totalBookings: number;
  totalActiveCourts: number;
  todaysRevenue: number;
  openBookingsCount: number;
  recentBookings: BookingRequest[];
}
